.github/actions_scripts/analyse_git_reference.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/python3
+
+#
+# on a triggered github workflow, this file does the decisions and variagles like
+#   - shall the build be released (otherwise just run builds to check if there are errors)
+#   - is it a prerelease
+#   - title, tag etc of release_tag
+#
+# see the last lines of the file to see what variables are set
+#
+
+
+import sys
+import os
+
+# get the jamulus version from the .pro file
+def get_jamulus_version(repo_path_on_disk):
+    jamulus_version = ""
+    with open (repo_path_on_disk + '/Jamulus.pro','r') as f:
+        pro_content = f.read()
+    pro_content = pro_content.replace('\r','')
+    pro_lines = pro_content.split('\n')
+    for line in pro_lines:
+        line = line.strip()
+        VERSION_LINE_STARTSWITH = 'VERSION = '
+        if line.startswith(VERSION_LINE_STARTSWITH):
+            jamulus_version = line[len(VERSION_LINE_STARTSWITH):]
+            return jamulus_version
+    return "UNKNOWN_VERSION"
+
+
+if len(sys.argv) == 1:
+   pass
+else:
+    print('wrong number of arguments')
+    print('Number of arguments:', len(sys.argv), 'arguments.')
+    print('Argument List:', str(sys.argv))
+    raise Exception("wrong agruments")
+    
+# derive workspace-path
+repo_path_on_disk = os.environ['GITHUB_WORKSPACE'] 
+
+# derive git related variables
+release_version_name = get_jamulus_version(repo_path_on_disk)
+fullref=os.environ['GITHUB_REF']
+reflist = fullref.split("/", 2)
+pushed_name = reflist[2]
+
+
+# run Changelog-script
+os.system('perl "{}"/.github/actions_scripts/getChangelog.pl "{}"/ChangeLog "{}" > "{}"/autoLatestChangelog.md'.format(
+    os.environ['GITHUB_WORKSPACE'],
+    os.environ['GITHUB_WORKSPACE'],
+    release_version_name,
+    os.environ['GITHUB_WORKSPACE']
+))
+
+# decisions about release, prerelease, title and tag
+if fullref.startswith("refs/tags/"):
+    print('this reference is a Tag')
+    release_tag = pushed_name # tag already exists
+    release_title="Release {}  ({})".format(release_version_name, pushed_name)
+    if pushed_name.startswith("r"):
+        publish_to_release = True
+        if "beta" in pushed_name:
+            print('this reference is a Beta-Release-Tag')
+            is_prerelease = True
+        else:
+            print('this reference is a Release-Tag')
+            is_prerelease = False
+    else:
+        print('this reference is a Non-Release-Tag')
+        publish_to_release = False
+        is_prerelease = True # just in case
+    
+        
+    if pushed_name == "latest":
+        print('this reference is a Latest-Tag')
```

Comment:
Why is the latest release a prerelease?

Reply:
I didnt care there, I thought the idea of this tag had died.

GitHub has a „latest release“ feature anyway.

Reply:
I’d there an URL which is always accessible and pointing to the latest release? I think @corrados said somewhere that probably many people use this somehow. I also think that a url which always points to the latest release is needed.

Reply:
look at this:
https://github.com/nefarius2001/jamulus/releases/latest

It points to the release with the youngest creation time, and is not related to the "latest" tag. I'd prefer that, since then the files also keep their version in name. But whoever actually does the releases, has the last word on how they should be done.

Reply:
Ok. I didn't know about this feature. 

> since then the files also keep their version in name

I would like to have a steady link which we could link on the jamulus.io page as (second) mirror.
The user would then click on "Download Jamulus" and the download would then start right away without leaving jamulus.io


This would provide a better user experience since they wouldn't be redirected to sourceforge which some people dislike (there are issues on that here).

The binaries would still be uploaded to SourceForge and maybe even linked on jamulus.io so that we have multiple mirrors. Multiple mirrors provide redundancy and also higher availability. If one is down, there is still the other one.